packages/mwp-app-route-plugin/src/handler.js

```diff
@@ -15,28 +15,29 @@ import url from 'url';
  *    the Vary header to 'User-Agent', in order for the google bots
  *    to crawl mobile and desktop versions of the site
  */
-export default (languageRenderers: { [string]: LanguageRenderer }) => (
+export default (languageRenderers: { [string]: LanguageRenderer }): any => (
 	request: HapiRequest,
-	reply: HapiReply
-) => {
+	h: HapiResponseToolkit
+): any => {
 	const pathname = request.getLangPrefixPath();
 	if (pathname !== request.url.pathname) {
-		return reply.redirect(url.format({ ...request.url, pathname }));
+		return h.redirect(url.format({ ...request.url, pathname }));
 	}
 	const requestLanguage = request.getLanguage();
 	const renderRequest = languageRenderers[requestLanguage];
 
-	renderRequest(request).then(
-		(renderResult: RenderResult) => {
+	return renderRequest(request).then(
+		(renderResult: RenderResult): any => {
```

Comment:
you don't have to declare a return value here - Flow will infer it and then check that it matches up with the wrapping function's declared return type.

technically, the type is

```js
(renderResult: RenderResult): Promise<HapiResponseToolkit | Error>
```

Although I think more generally Hapi is okay with a response type like this:

```js
type HapiHandlerSyncResponse = HapiResponseToolkit | Error | string;
type Response = HapiHandlerSyncResponse | Promise<HapiHandlerSyncResponse>;
```

because I think you can return a string, Error, or `h.response()` directly?

Reply:
You can return string, object, hapi response object (`h.reponse()`, `h.redirect()`, etc.), and maybe an Error (not sure about Error). But this works:

```
declare type HapiHandlerReturnValue = HapiResponseObject | Error | string | { [string]: any };
declare type HapiHandler = (HapiRequest, HapiResponseToolkit) => HapiHandlerReturnValue | Promise<HapiHandlerReturnValue>;
```
Thanks for the flow help!